### PR TITLE
Added evaluation metrics

### PR DESCRIPTION
**Pull request by anna-kartynnik**
Right now I ran the evaluation from the terminal having NN saved as file and getting predictions after loading the NN. If you want to run the evaluation from NN trainer, please let me know.

### CODE REVIEW

**Comment by niharmaheshwari** [comment]
Hi @anna-kartynnik , @ArjunAhuja-git has added many generalizations to the neural network code. Should we run and save the neural network output against this code?

File: graph-builder/data_reader.py [hunk 1]
```diff
@@ -2,38 +2,65 @@
...
+		# More details about the file format can be found at http://yann.lecun.com/exdb/mnist/.
+		with gzip.open(abs_file_path, "r") as f:
+			# First 4 bytes is a magic number.
+			magic_number = int.from_bytes(f.read(4), "big")
+			# Second 4 bytes is the number of items.
+			number_of_items = int.from_bytes(f.read(4), "big")
+			# Rest is the labels from 0 to 9.
+			label_data = f.read()
+			labels = np.frombuffer(label_data, dtype=np.uint8)
+			return labels
```
Comment (ArjunAhuja-git) [inline graph-builder/data_reader.py:48]: Duplicate?

**Review by ArjunAhuja-git — APPROVED** [review]
Looks good, let's make sure its not tied to any specific implementation(Robust to be used with LSH, annoy and others) I kindof feel there is still a bit of coupling between Neural LSH and metrics.

File: graph-builder/data_reader.py (hunk 1 above)
Comment (anna-kartynnik) [reply graph-builder/data_reader.py:48]: No, this method reads MNIST **labels**. Images and labels have different formats. More details are here http://yann.lecun.com/exdb/mnist/

**Comment by anna-kartynnik** [comment]
The idea is to use `metrics_utils.py` from different algorithms. `evaluation.py` is an example of usage the functions from `metrics_utils` for Neural-LSH specifically.
I'm refactoring `get_average_intrinsic_accuracy` to be more independent. Other functions don't depend on the details of Neural LSH.

**Pull request merged**